### ISSUE TITLE
Cannot create FlannBasedMatcher with parameters

### ISSUE THREAD

**Issue by MightyAlex200** [twistedfall/opencv-rust#155]
`FlannBasedMatcher::new` requires `Ptr<_>` versions of `IndexParams` and `SearchParams`, which do not appear to have constructor functions that return `Ptr`s. I don't know very much about C++, but I tried creating new `Ptr` objects from the pointers of the index and search parameters, and this caused a segfault:
```rust
let mut index_params = IndexParams::default().unwrap();
index_params.set_algorithm(FLANN_INDEX_KDTREE).unwrap();
index_params.set_int("trees", 5).unwrap();
let mut search_params = SearchParams::new(50, 0., true).unwrap();
let index_params = unsafe { Ptr::from_raw(index_params.as_raw_mut_IndexParams()) };
let search_params = unsafe { Ptr::from_raw(search_params.as_raw_mut_SearchParams()) };

let flann = FlannBasedMatcher::new(&index_params, &search_params); // segfault
```

**Comment by twistedfall** [comment]
Thanks! This is definitely a deficiency of the API, I’ll see how this can be fixed.

**Comment by twistedfall** [comment]
Can you please check out the `ptr` branch and tell me if it works for you? There is a `new` method of non-abstract inner types for `Ptr`.

**Comment by MightyAlex200** [comment]
Looks like it's working perfectly now. Thank you for the very quick response! Should I keep this issue open while you merge it?

**Comment by twistedfall** [comment]
Yes, please, I’ll close it when the feature is released

**Comment by twistedfall** [comment]
This is now released in version 0.44.0
